src/modules/auth/redux/actions.ts

```diff
@@ -18,3 +19,125 @@ export const authActions = {
     );
   },
 };
+
+export const setCurrentUser = (currentUser) => ({
+  type: AuthConsts.SET_CURRENT_USER,
+  payload: {
+    currentUser 
+  }, 
+})
+
+export const getCurrentUser = () => async (dispatch: Dispatch) => {
+  try {
+    const currentUser = await AuthService.getCurrentUser();
+
+    dispatch(setCurrentUser(currentUser));
+  }catch(error) {
+    console.error(error);
+    message.error('Unkown Error', error);
+  }
+}
+
+export const setAuthState = (state) => ({
+  type: AuthConsts.SET_AUTH_STATE,
+  payload: {
+    state, 
+  }
+})
+
+export const setAuthData = (data) => ({
+  type: AuthConsts.SET_AUTH_DATA,
+  payload: {
+    data, 
+  }
+})
+
+const resolveAuthFromStorage = () => {
+  const localSession = localStorage.getItem(SESSION_KEY) || '';
```

Comment:
add `if(!process.browser) return {}` prevent error

Reply:
useEffect wont get run in next server. so it will not cause error.